tests/PendingRequestMixinTest.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+use GuzzleHttp\HandlerStack;
+use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Support\Facades\Http;
+use Onlime\LaravelHttpClientGlobalLogger\Mixins\PendingRequestMixin;
+
+it('rebuilds the guzzle handler stack to include the logger middleware at the bottom of the stack', function () {
+    PendingRequest::mixin(new PendingRequestMixin);
+
+    /** @var PendingRequest $pendingRequest */
+    $pendingRequest = Http::withHeaders(['X-Test' => 'true'])->log();
+
+    /** @var HandlerStack $handler */
+    $handler = $pendingRequest->getOptions()['handler'];
+    expect($handler)->toBeInstanceOf(HandlerStack::class);
+
+    // It adds a logger for each format (see config('http-client-global-logger.format'))
+    expect($handler->hasHandler('http-client-global-logger-0'))->toBeTrue()
```

Comment:
@pascalbaljet `hasHandler()` does not expect middleware name. Shouldn't it be something like:

```php
    expect($handler)->toBeInstanceOf(HandlerStack::class)
        ->and($handler->hasHandler())->toBeTrue()
        // It adds a logger for each format (see config('http-client-global-logger.format'))
        ->and($handler->findByName('http-client-global-logger-request'))->toBeInt()
        ->and($handler->findByName('http-client-global-logger-response'))->toBeInt();
```

but `HandlerStack::findByName()` is `private`, so currently don't know how to check that. Also, the keys are `request|response` in default config.

Reply:
Another option would be to use something like `spatie/invade` and check the private `$stack` property:

```php
$stack = invade($handler)->stack;
$middlewareNames = Arr::pluck($stack, 1);

expect($middlewareNames)->toContain('http-client-global-logger-0')
    ->and($middlewareNames)->toContain('http-client-global-logger-1')
    ->and($middlewareNames)->not->toContain('http-client-global-logger-2');
```

Reply:
yes, we all love `spatie/invade`! 😄 prefer this solution

Reply:
Fixed in 51d1c9c

Reply:
I have simplified it further in 7fbada6 by invading `HandlerStack::findByName()`